src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/AStar.java

```diff
@@ -0,0 +1,86 @@
+package edu.greenblitz.tobyDetermined.Nodesssss;
+
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.utils.GBMath;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+
+import static edu.greenblitz.tobyDetermined.RobotMap.TelescopicArm.PresetPositions.*;
+import static edu.greenblitz.tobyDetermined.RobotMap.TelescopicArm.PresetPositions;
+
+public class AStar {
+
+    public static double getDistanceToStartPlusEnd(PresetPositions current, NodeArm start, NodeArm end) {
+        double gCost = Math.abs(GBMath.distance(GBMath.convertToCartesian(start.getExtendPos() + RobotMap.TelescopicArm.Extender.STARTING_LENGTH, start.getAnglePos()), GBMath.convertToCartesian(NodeBase.getNode(current).getExtendPos() + RobotMap.TelescopicArm.Extender.STARTING_LENGTH, NodeBase.getNode(current).getAnglePos())));
+        double hCost = Math.abs(GBMath.distance(GBMath.convertToCartesian(end.getExtendPos() + RobotMap.TelescopicArm.Extender.STARTING_LENGTH, end.getAnglePos()), GBMath.convertToCartesian(NodeBase.getNode(current).getExtendPos() + RobotMap.TelescopicArm.Extender.STARTING_LENGTH, NodeBase.getNode(current).getAnglePos())));
+        return gCost + hCost;
+    }
+
+
+    public static PresetPositions getLowestFcost(LinkedList<PresetPositions> open, NodeArm start, NodeArm end) {
+        int saveI = 0;
+        double fCost = getDistanceToStartPlusEnd(open.get(0), start, end);
+        for (int i = 1; i < open.size(); i++) {
+            double currentFCost = getDistanceToStartPlusEnd(open.get(i), start, end);
+            if (currentFCost < fCost) {
+                fCost = currentFCost;
+                saveI = i;
+            }
+        }
+        return open.get(saveI);
+    }
+
+
+    public static boolean isInList(PresetPositions nodeArmPos, LinkedList<PresetPositions> list) {
+        return list.contains(nodeArmPos);
+    }
+
+    public static LinkedList<PresetPositions> returnPath(PresetPositions nodeArm, Map<PresetPositions, PresetPositions> parents) {
+        LinkedList<PresetPositions> pathList = new LinkedList<>();
+        PresetPositions current = nodeArm;
+        while (current != null) {
+            pathList.addFirst(current);
+            current = parents.get(current);
+        }
+        printPath(pathList);
+        return pathList;
+    }
+
+    public static void printPath(LinkedList<PresetPositions> pathList) {
+        for (int i = 0; i < pathList.size(); i++) {
+            System.out.print(pathList.get(i) + ", ");
+        }
+        System.out.println();
+    }
+
+    public static LinkedList<PresetPositions> getPath(PresetPositions start, PresetPositions end) {
```

Comment:
the difference isnt clear between getPath() and returnPath() by their names. maybe change the 'get' to 'calculate' or smth